### DISCUSSION TITLE
How to use Keycloak for authentication with OAuthProvider

### DISCUSSION THREAD

**Discussion by thinkORo** [flet-dev/flet#2587] [Q&A]
### Question

I'm using Keycloak as IDP for FastAPI and others. And now I'd like use it to authenticate the user in Flet as well?

I tried to follow this post/documentation: https://flet.dev/docs/guides/python/authentication/#configuring-a-custom-oauth-provider
But I didn't get it running.

Has anyone ever tried to get this combination (Flet + Keycloak) to work?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by thinkORo** [comment]
I'm sorry but everything works fine with the provided example in the documentation.
It was my fault as I didn't catch any value from the result and therefore I didn't see any result.
Works like a charm.

**Reply by jersonjunior** [reply]
Could you provide the code?

**Reply by thinkORo** [reply]
Did you try the code snippet from the website I referenced in my first comment? If yes, what the error?

My code is very, very similar but slightly different because of my keycloak setup.

I'd assume that my code wouldn't help you as your keycloak setup is probably different.

**Reply by jersonjunior** [reply]
I was successful, but I can't get the user's information after logging in

**Reply by thinkORo** [reply]
My understanding: there is no userinfo in Flet's OAuthProvider.

But ... you can use the provided token (`access_token = router.page.auth.token.access_token`) to ask your OAuthProvider (in my case keycloak) to provide the required userinfo. Like so:

```
from keycloak import KeycloakOpenID

keycloak_openid = KeycloakOpenID(
    server_url="your keycloak sever",
    realm_name="your realm",
    client_id="your client id of resective realm",
    client_secret_key="the corresponding secret key")

userinfo = keycloak_openid.userinfo(access_token)
username = userinfo["name"]
```

Hint: You have to install the python keycloak component with pip like to: `pip install python-keycloak`

I hope that makes sense to you? And I hope that works to you.

If not drop a comment and we will find a way to make it work.
